Testing: TestPipelineExceedMaximumServing

```
--- FAIL: TestPipelineExceedMaximumServing (0.01s)
	pipeline_test.go:100: failed to send out message
2016-09-09 22:04:30.068434 I | rafthttp: started HTTP pipelining with peer 1
2016-09-09 22:04:30.079030 I | rafthttp: stopped HTTP pipelining with peer 1
2016-09-09 22:04:30.079704 I | rafthttp: started HTTP pipelining with peer 1
2016-09-09 22:04:30.090107 I | rafthttp: stopped HTTP pipelining with peer 1
2016-09-09 22:04:30.090789 I | rafthttp: started HTTP pipelining with peer 1
2016-09-09 22:04:30.091157 I | rafthttp: stopped HTTP pipelining with peer 1
2016-09-09 22:04:30.091562 I | rafthttp: started HTTP pipelining with peer 1
2016-09-09 22:04:30.091932 I | rafthttp: stopped HTTP pipelining with peer 1
2016-09-09 22:04:30.092380 I | rafthttp: started HTTP pipelining with peer 1
2016-09-09 22:04:30.092741 I | rafthttp: stopped HTTP pipelining with peer 1
2016-09-09 22:04:30.093341 I | rafthttp: started HTTP pipelining with peer 1
2016-09-09 22:04:30.093703 I | rafthttp: stopped HTTP pipelining with peer 1
2016-09-09 22:04:30.094310 I | rafthttp: started HTTP pipelining with peer 1
2016-09-09 22:04:30.097785 I | rafthttp: stopped HTTP pipelining with peer 1
2016-09-09 22:04:30.098431 I | rafthttp: start to send database snapshot [index: 0, to 1]...
2016-09-09 22:04:30.099874 I | rafthttp: receiving database snapshot [index:0, from 0] ...
2016-09-09 22:04:30.100052 I | snap: saved database snapshot to disk [total bytes: 5]
2016-09-09 22:04:30.100117 I | rafthttp: received and saved database snapshot [index: 0, from: 0] successfully
2016-09-09 22:04:30.100379 I | rafthttp: peer 1 became active
2016-09-09 22:04:30.100414 I | rafthttp: database snapshot [index: 0, to: 1] sent out successfully
2016-09-09 22:04:30.100909 I | rafthttp: start to send database snapshot [index: 0, to 1]...
2016-09-09 22:04:30.101732 W | rafthttp: database snapshot [index: 0, to: 1] failed to be sent out (ioutil: short read)
2016-09-09 22:04:30.101803 I | rafthttp: receiving database snapshot [index:0, from 0] ...
2016-09-09 22:04:30.101919 E | rafthttp: failed to save KV snapshot (unexpected EOF)
2016-09-09 22:04:30.102497 I | rafthttp: start to send database snapshot [index: 0, to 1]...
2016-09-09 22:04:30.103658 I | rafthttp: receiving database snapshot [index:0, from 0] ...
2016-09-09 22:04:30.103729 W | rafthttp: database snapshot [index: 0, to: 1] failed to be sent out (ioutil: short read)
2016-09-09 22:04:30.103817 E | rafthttp: failed to save KV snapshot (unexpected EOF)
2016-09-09 22:04:30.104227 I | rafthttp: start to send database snapshot [index: 0, to 1]...
2016-09-09 22:04:30.105222 W | rafthttp: database snapshot [index: 0, to: 1] failed to be sent out (ioutil: expect EOF)
2016-09-09 22:04:30.105363 I | rafthttp: receiving database snapshot [index:0, from 0] ...
2016-09-09 22:04:30.105466 E | rafthttp: failed to save KV snapshot (unexpected EOF)
2016-09-09 22:04:30.106642 I | rafthttp: started streaming with peer 1 (writer)
2016-09-09 22:04:30.106719 I | rafthttp: peer 1 became active
2016-09-09 22:04:30.106760 I | rafthttp: established a TCP streaming connection with peer 1 (stream Message writer)
2016-09-09 22:04:30.107668 W | rafthttp: closed an existing TCP streaming connection with peer 1 (stream Message writer)
2016-09-09 22:04:30.107710 I | rafthttp: established a TCP streaming connection with peer 1 (stream Message writer)
2016-09-09 22:04:30.108795 W | rafthttp: closed an existing TCP streaming connection with peer 1 (stream Message writer)
2016-09-09 22:04:30.108866 I | rafthttp: established a TCP streaming connection with peer 1 (stream Message writer)
2016-09-09 22:04:30.127550 I | rafthttp: closed the TCP streaming connection with peer 1 (stream Message writer)
2016-09-09 22:04:30.127627 I | rafthttp: stopped streaming with peer 1 (writer)
2016-09-09 22:04:30.130066 I | rafthttp: started streaming with peer 1 (writer)
2016-09-09 22:04:30.130168 I | rafthttp: peer 1 became active
2016-09-09 22:04:30.130227 I | rafthttp: established a TCP streaming connection with peer 1 (stream Message writer)
2016-09-09 22:04:30.130334 E | rafthttp: failed to write 1 on stream Message (blah)
2016-09-09 22:04:30.130375 I | rafthttp: peer 1 became inactive
2016-09-09 22:04:30.131429 W | rafthttp: lost the TCP streaming connection with peer 1 (stream Message writer)
2016-09-09 22:04:30.131522 I | rafthttp: stopped streaming with peer 1 (writer)
2016-09-09 22:04:30.155729 I | rafthttp: started streaming with peer 2 (stream Message reader)
2016-09-09 22:04:30.166170 I | rafthttp: peer 2 became active
2016-09-09 22:04:30.166255 I | rafthttp: established a TCP streaming connection with peer 2 (stream Message reader)
2016-09-09 22:04:30.166334 W | rafthttp: lost the TCP streaming connection with peer 2 (stream Message reader)
2016-09-09 22:04:30.166527 I | rafthttp: stopped streaming with peer 2 (stream Message reader)
2016-09-09 22:04:30.189377 I | rafthttp: started streaming with peer 2 (stream Message reader)
2016-09-09 22:04:30.189660 I | rafthttp: started streaming with peer 1 (writer)
2016-09-09 22:04:30.190654 I | rafthttp: peer 1 became active
2016-09-09 22:04:30.190717 I | rafthttp: established a TCP streaming connection with peer 1 (stream Message writer)
2016-09-09 22:04:30.190765 I | rafthttp: peer 2 became active
2016-09-09 22:04:30.190799 I | rafthttp: established a TCP streaming connection with peer 2 (stream Message reader)
2016-09-09 22:04:30.191176 W | rafthttp: lost the TCP streaming connection with peer 2 (stream Message reader)
2016-09-09 22:04:30.191247 I | rafthttp: stopped streaming with peer 2 (stream Message reader)
2016-09-09 22:04:30.194575 I | rafthttp: started streaming with peer 1 (writer)
2016-09-09 22:04:30.197213 I | rafthttp: started streaming with peer 2 (stream Message reader)
2016-09-09 22:04:30.199112 I | rafthttp: peer 1 became active
2016-09-09 22:04:30.199183 I | rafthttp: established a TCP streaming connection with peer 1 (stream Message writer)
2016-09-09 22:04:30.199371 I | rafthttp: peer 2 became active
2016-09-09 22:04:30.199426 I | rafthttp: established a TCP streaming connection with peer 2 (stream Message reader)
2016-09-09 22:04:30.199917 W | rafthttp: lost the TCP streaming connection with peer 2 (stream Message reader)
2016-09-09 22:04:30.199991 I | rafthttp: stopped streaming with peer 2 (stream Message reader)
2016-09-09 22:04:30.205364 I | rafthttp: started streaming with peer 1 (writer)
2016-09-09 22:04:30.207941 I | rafthttp: started streaming with peer 2 (stream MsgApp v2 reader)
2016-09-09 22:04:30.209955 I | rafthttp: peer 1 became active
2016-09-09 22:04:30.210246 I | rafthttp: established a TCP streaming connection with peer 1 (stream MsgApp v2 writer)
2016-09-09 22:04:30.210473 I | rafthttp: peer 2 became active
2016-09-09 22:04:30.210542 I | rafthttp: established a TCP streaming connection with peer 2 (stream MsgApp v2 reader)
2016-09-09 22:04:30.211930 W | rafthttp: lost the TCP streaming connection with peer 2 (stream MsgApp v2 reader)
2016-09-09 22:04:30.212032 I | rafthttp: stopped streaming with peer 2 (stream MsgApp v2 reader)
2016-09-09 22:04:30.223443 I | rafthttp: closed the TCP streaming connection with peer 1 (stream MsgApp v2 writer)
2016-09-09 22:04:30.223546 I | rafthttp: stopped streaming with peer 1 (writer)
2016-09-09 22:04:30.224670 I | rafthttp: closed the TCP streaming connection with peer 1 (stream Message writer)
2016-09-09 22:04:30.224751 I | rafthttp: stopped streaming with peer 1 (writer)
2016-09-09 22:04:30.227687 I | rafthttp: closed the TCP streaming connection with peer 1 (stream Message writer)
2016-09-09 22:04:30.227753 I | rafthttp: stopped streaming with peer 1 (writer)
2016-09-09 22:04:30.228946 I | rafthttp: starting peer 1...
2016-09-09 22:04:30.229324 I | rafthttp: started HTTP pipelining with peer 1
2016-09-09 22:04:30.231999 I | rafthttp: started streaming with peer 1 (writer)
2016-09-09 22:04:30.235914 I | rafthttp: started streaming with peer 1 (writer)
2016-09-09 22:04:30.250668 I | rafthttp: started streaming with peer 1 (stream MsgApp v2 reader)
2016-09-09 22:04:30.251175 I | rafthttp: started streaming with peer 1 (stream Message reader)
2016-09-09 22:04:30.251446 I | rafthttp: started peer 1
2016-09-09 22:04:30.251751 I | rafthttp: added peer 1
2016-09-09 22:04:30.251898 I | rafthttp: stopping peer 1...
2016-09-09 22:04:30.252037 I | rafthttp: stopped streaming with peer 1 (writer)
2016-09-09 22:04:30.252157 I | rafthttp: stopped streaming with peer 1 (writer)
2016-09-09 22:04:30.252308 I | rafthttp: stopped HTTP pipelining with peer 1
2016-09-09 22:04:30.252424 I | rafthttp: stopped streaming with peer 1 (stream MsgApp v2 reader)
2016-09-09 22:04:30.252521 I | rafthttp: stopped streaming with peer 1 (stream Message reader)
2016-09-09 22:04:30.252590 I | rafthttp: stopped peer 1
2016-09-09 22:04:30.252990 I | rafthttp: starting peer 1...
--
FAIL
coverage: 76.1% of statements
```

--- a/rafthttp/pipeline_test.go
+++ b/rafthttp/pipeline_test.go
@@ -67,9 +67,9 @@ func TestPipelineKeepSendingWhenPostError(t *testing.T) {
 }
 
 func TestPipelineExceedMaximumServing(t *testing.T) {
-	tr := newRoundTripperBlocker()
+	rt := newRoundTripperBlocker()
 	picker := mustNewURLPicker(t, []string{"http://localhost:2380"})
-	tp := &Transport{pipelineRt: tr}
+	tp := &Transport{pipelineRt: rt}
 	p := startTestPipeline(tp, picker)
 	defer p.stop()
 
@@ -91,12 +91,12 @@ func TestPipelineExceedMaximumServing(t *testing.T) {
 	}
 
 	// unblock the senders and force them to send out the data
-	tr.unblock()
+	rt.unblock()
 
 	// It could send new data after previous ones succeed
 	select {
 	case p.msgc <- raftpb.Message{}:
-	case <-time.After(10 * time.Millisecond):
+	case <-time.After(time.Second):
 		t.Errorf("failed to send out message")
 	}
 }

